.github/workflows/deploy.yml

```
@@ -0,0 +1,46 @@
+        password: ${{ secrets.DOCKER_PASSWORD }}
+
+    # 3. Build Docker image
+    - name: Build Docker image
+      run: |
+        IMAGE_NAME="${{ secrets.DOCKER_USERNAME }}/ma-yone:${{ github.sha }}"
+        docker build -t $IMAGE_NAME .
+        echo "IMAGE_NAME=$IMAGE_NAME" >> $GITHUB_ENV
+
+    # 4. Push Docker image to Docker Hub
+    - name: Push Docker image
+      run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:13: Double quote to prevent globbing and word splitting

(shellcheck)